### ISSUE TITLE
[AdHoc] Midway Arcade Treasures: Extended Play

### ISSUE THREAD

**Issue by TemporaryPSP** [hrydgard/ppsspp#15177]
### Game or games this happens in

ULUS-10059 Midway Arcade Treasures: Extended Play

### What area of the game / PPSSPP

AdHoc partially works with some games (localhost only)
I tested the 3 Mortal Kombat games and Gauntlet (a 4 player game)
MK1: 2nd instance disconnects as soon as the loading screen appears
MK2: 2nd instance disconnects in the middle of loading
MK3: 2nd instance disconnects but immediately reconnects, playable
Gauntlet: all 4 instances don't encounter any issue

I tried using a 2nd device (an Android phone) and it's just messy. Both devices would connect and disconnect even in the game select screen and I can't even get them to meet in the game lobby unlike localhost.

### What should happen

AdHoc should work.

### Platform

Windows

### Mobile phone model or graphics card

Samsung Galaxy A51

### PPSSPP version affected

v1.12.3

### Last working version

Don't think it has ever worked before.

### Graphics backend (3D API)

Direct3D 9

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [X] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

**Comment by ANR2ME** [comment]
Are those games classic games like one of those capcom games?

**Comment by TemporaryPSP** [comment]
Yes, it's a collection of their old arcade titles, dating as far back as 1980.

**Comment by ANR2ME** [comment]
MK3 seems to be using PDP socket with 20ms as timeout, is your ping to the Android device higher than 20ms?

**Comment by TemporaryPSP** [comment]
I can't get any game to work on multiple devices. Only localhost works, it just kicks the "joiner" out of the host's lobby every time.

**Comment by ANR2ME** [comment]
Might be because localhost have less than 1ms latency that's why it works on most games, while the `ping` from my laptop to my Android 11 phone can get more than 100ms latency LOL the phone was tethered to my laptop so not sure why the ping is so high like that.

**Comment by TemporaryPSP** [comment]
I checked and the average latency I got was around 75-90 ms.

**Comment by ANR2ME** [comment]
You can try changing the Minimum Timeout on Networking settings to the highest latency you've seen.

**Comment by TemporaryPSP** [comment]
With that setting, the lobby doesn't disband as frequently anymore it seems? Unfortunately neither of the MK games work. They still disconnect upon loading, MK3 got so close but the START button doesn't work for both devices for some reason.

**Comment by TemporaryPSP** [comment]
Disregard everything I said about 2nd device being messy, for some reason it works just like localhost now except getting kicked from the lobby once in a while. That setting didn't do much for me, MK3 works well but not the other 2.
